TutorLizard.BusinessLogic/Services/TutorService.cs

```diff
@@ -14,103 +14,89 @@ public TutorService(ITutorDataAccess dataAccess, IUserIdentityService userIdenti
+        return _dataAccess.CreateScheduleItem(adId, dateTime);
     }
-    public Ad? GetAdById(int adId)
+    public Ad? GetAdById(int? adId)
     {
         // return Ad (from _dataAccess) with provided adId
         // return null if no such Ad
-        return new Ad()
+        adId = _dataAccess.GetAllAdsId();
+        if (adId is null)
         {
-            // this is only for tests
-            Id = adId,
-            Title = "Test"
-        };
+            return null;
+        }
+        return _dataAccess.GetAdById((int)adId);
     }
-    public ScheduleItem? GetScheduleItemById(int scheduleItemId)
+    public ScheduleItem? GetScheduleItemById(int? scheduleItemId)
     {
-        return new ScheduleItem()
+        scheduleItemId = _dataAccess.GetScheduleItemId();
+        if (scheduleItemId is null)
         {
-            // this is only for tests
-            Id = scheduleItemId,
-            DateTime = DateTime.Now.AddHours(1),
-        };
+            return null;
+        }
+        return _dataAccess.GetScheduleItemById(scheduleItemId);
     }
     public List<Ad> GetUsersAds()
     {
-        return [
-            // this is only for tests
-            GetAdById(1)!,
-            GetAdById(2)!,
-            GetAdById(3)!
-            ];
+        List<Ad> ads = new List<Ad>();
+        return [.. _dataAccess.GetUsersAds()];
     }
     public List<ScheduleItem> GetUsersScheduleItems()
     {
-        return [
-            // this is only for tests
-            GetScheduleItemById(1)!,
-            GetScheduleItemById(2)!,
-            GetScheduleItemById(3)!
-            ];
+        return _dataAccess.GetUsersScheduleItems();
     }
     public List<AdRequest> GetUsersAdRequests()
     {
-        return [
-            // this is only for tests
-            new AdRequest()
-            {
-                Id = 1,
-                AdId = 1,
-                IsAccepted = false,
-                Message = "Test Ad Request",
-                StudentId = 1
-            },
-        ];
+        return _dataAccess.GetUsersAdRequests();
     }
     public List<ScheduleItemRequest> GetUsersScheduleItemRequests()
     {
-        return [
-            new ScheduleItemRequest()
-            {
-                // this is only for tests
-                Id = 1,
-                IsAccepted = false,
-                ScheduleItemId = 1,
-                UserId = 1
-            }];
+        return _dataAccess.GetUsersScheduleItemRequests();
     }
     public string GetStudentUserNameByAdRequestId(int adRequestId)
     {
-        // this is only for tests
-        return "TestStudentName1";
+        AdRequest adRequest = _dataAccess.GetAdRequestById(adRequestId);
+        if (adRequest != null)
+        {
+            int studentId = adRequest.StudentId;
+            User user = _userIdentityService.GetUserById(studentId);
+            if (user != null)
+            {
+                return user.Name;
+            }
+        }
+        return null;
```

Comment:
I didn't make the return type nullable, so let's not return null. Return empty string (`""`) instead.